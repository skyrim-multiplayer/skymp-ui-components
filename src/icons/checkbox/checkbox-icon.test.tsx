import React from "react";
import { shallow, mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";

import { CheckboxIcon } from "./index";

describe("CheckboxIcon", () => {
  it("renders without errors", () => {
    shallow(<CheckboxIcon />);
  });

  it("`isChecked` prop is true", () => {
    const wrapper = mount(<CheckboxIcon isChecked={true} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it("`isChecked` prop is false", () => {
    const wrapper = mount(<CheckboxIcon isChecked={false} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
