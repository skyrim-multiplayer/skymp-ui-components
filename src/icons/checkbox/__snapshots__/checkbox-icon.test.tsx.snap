// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxIcon \`isChecked\` prop is false 1`] = `
<Component
  isChecked={false}
>
  <Component
    strokeMiterlimit="22.9256"
    strokeWidth="0.05"
    viewBox="0 0 0.61 0.63"
  >
    <svg
      height="1em"
      strokeMiterlimit="22.9256"
      strokeWidth="0.05"
      style={
        Object {
          "clipRule": "evenodd",
          "fillRule": "evenodd",
          "shapeRendering": "geometricPrecision",
          "textRendering": "geometricPrecision",
        }
      }
      viewBox="0 0 0.61 0.63"
      width="1em"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <g>
        <rect
          fill="none"
          height="0.54"
          rx="0.11"
          ry="0.11"
          stroke="currentColor"
          width="0.54"
          x="0.03"
          y="0.06"
        />
      </g>
    </svg>
  </Component>
</Component>
`;

exports[`CheckboxIcon \`isChecked\` prop is true 1`] = `
<Component
  isChecked={true}
>
  <Component
    strokeMiterlimit="22.9256"
    strokeWidth="0.05"
    viewBox="0 0 0.61 0.63"
  >
    <svg
      height="1em"
      strokeMiterlimit="22.9256"
      strokeWidth="0.05"
      style={
        Object {
          "clipRule": "evenodd",
          "fillRule": "evenodd",
          "shapeRendering": "geometricPrecision",
          "textRendering": "geometricPrecision",
        }
      }
      viewBox="0 0 0.61 0.63"
      width="1em"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <g>
        <rect
          fill="none"
          height="0.54"
          rx="0.11"
          ry="0.11"
          stroke="currentColor"
          width="0.54"
          x="0.03"
          y="0.06"
        />
        <path
          d="M0.31 0.37l-0.1 -0.11c-0.01,-0.01 -0.01,-0.01 -0.01,-0.01l-0.04 -0.05c-0.03,-0.04 -0.09,0.01 -0.05,0.05 0,0 0.01,0.01 0.01,0.01l0.1 0.11c0.01,0.01 0.01,0.01 0.02,0.02 0,0 0.01,0.01 0.01,0.01l0.03 0.04c0.01,0.01 0.02,0.02 0.03,0.03 0.02,0.01 0.04,-0 0.05,-0.02 0.01,-0.02 0.02,-0.03 0.03,-0.05 0.01,-0.01 0.02,-0.03 0.03,-0.04 0.01,-0.02 0.02,-0.03 0.03,-0.05l0.15 -0.22c0.01,-0.02 0.03,-0.03 0.03,-0.05 0,-0.03 -0.03,-0.04 -0.05,-0.03 -0.02,0.01 -0.03,0.03 -0.04,0.05 -0.01,0.02 -0.02,0.03 -0.03,0.05l-0.18 0.27z"
          fill="currentColor"
          stroke="none"
        />
      </g>
    </svg>
  </Component>
</Component>
`;
